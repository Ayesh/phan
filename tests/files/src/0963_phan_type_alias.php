<?php

/**
 * NOTE: Due to a limitation of the php-ast parser, 'phan-type' annotations must be on element doc comments such as classlikes, functions, etc.
 * @phan-type UserData = array{name:string, id:int}
 */
class UserModel {
    /**
     * @var UserData
     */
    private $data;
    /**
     * @param UserData $data
     */
    public function __construct(array $data) {
        $this->data = $data;
    }

    /** @return UserData */
    public function getData(): array {
        return $this->data;
    }
}

// should warn about 'id'
$value = new UserModel(['name' => 'test', 'id' => 'invalid']);
$data = $value->getData();

'@phan-debug-var $data';
