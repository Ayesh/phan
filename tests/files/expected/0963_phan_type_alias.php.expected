%s:26 PhanTypeMismatchArgument Argument 1 ($data) is ['name'=>'test','id'=>'invalid'] of type array{name:'test',id:'invalid'} but \UserModel::__construct() takes array{name:string,id:int} defined at %s:15
%s:27 PhanDebugAnnotation @phan-debug-var requested for variable $data - it has union type array{name:string,id:int}(real=array)
